src/coreclr/jit/optcse.cpp

```diff
@@ -4016,12 +4072,12 @@ void CSE_Heuristic::Initialize()
         }
 
 #ifdef TARGET_X86
-        // Treat floating point and 64 bit integers as always on the stack
-        if (varTypeIsFloating(varDsc->TypeGet()) || varTypeIsLong(varDsc->TypeGet()))
```

Comment:
This (for floating-point) is a consideration that hasn't been true for years. We've required and preferred using the SSE/SSE2 registers for years on x86 and only use the x87 stack space where required. So we really want to just treat it like any other enregisterable value, same as on x64.